source/parser.hera

```diff
@@ -3513,7 +3514,9 @@ RestoreAll
 ExpressionStatement
   # NOTE: semi-colons are being handled elsewhere
   # NOTE: Shouldn't need negative lookahead if shadowed in the proper order
-  Expression
+  # NOTE: Using ExtendedExpression instead of Expression to allow for
+  # e.g. await forms of IterationExpressions that can't be IterationStatement
+  ExtendedExpression
```

Comment:
This change will forbid `a,b` expressions pretty much everywhere. I'm not necessarily against it just want to note that it's a bigger change than it appears.

Reply:
Adding to `Expression` is probably the minimally invasive change.

Reply:
Actually maybe inside `AssignmentExpression` adjacent to`PipelineExpression`.

Reply:
Looks good!